biggerquery/build.py

```diff
@@ -0,0 +1,292 @@
+import os
+from pathlib import Path
+import subprocess
+import shutil
+import distutils.cmd
+from datetime import datetime
+import setuptools
+import unittest
+
+import xmlrunner
+
+from .cli import walk_workflows, import_deployment_config, _valid_datetime
+from .dagbuilder import generate_dag_file
+from .resources import read_requirements, find_all_resources
+from .utils import resolve
+from .version import get_version
+from .utils import run_process
+
+
+__all__ = [
+    'project_setup',
+    'auto_configuration'
+]
+
+
+def now(template: str = "%Y-%m-%d %H:00:00"):
+    return datetime.now().strftime(template)
+
+
+def run_tests(build_dir: Path, test_package: Path):
+    runner = xmlrunner.XMLTestRunner(output=resolve(build_dir / 'junit-reports'))
+    suite = unittest.TestLoader().discover(start_dir=resolve(test_package))
+    result = runner.run(suite)
+    if result.errors:
+        raise ValueError('Test suite failed.')
+
+
+def clear_image_leftovers(image_dir: Path):
+    print(f'Removing: {str(image_dir.absolute())}')
+    shutil.rmtree(image_dir, ignore_errors=True)
+
+
+def clear_dags_leftovers(dags_dir: Path):
+    print(f'Removing: {str(dags_dir.absolute())}')
+    shutil.rmtree(dags_dir, ignore_errors=True)
+
+
+def clear_package_leftovers(dist_dir: Path, eggs_dir: Path, build_dir: Path):
+    for to_delete in [build_dir, dist_dir, eggs_dir]:
+        print(f'Removing: {str(to_delete.absolute())}')
+        shutil.rmtree(to_delete, ignore_errors=True)
+
+
+def get_docker_image_id(tag):
+    images = subprocess.getoutput(f"docker images -q {tag}")
+    return images.split('\n')[0]
+
+
+def remove_docker_image_from_local_registry(tag):
+    print('Removing image from the local registry')
+    run_process(f"docker rmi {get_docker_image_id(tag)}")
+
+
+def export_docker_image_to_file(tag: str, target_dir: Path, version: str):
+    print(f'Exporting image to a file')
+    image_target_path = target_dir / f'image-{version}.tar'
+    run_process(f"docker image save {get_docker_image_id(tag)} -o {resolve(image_target_path)}")
+
+
+def build_docker_image(project_dir: Path, tag: str):
+    print('Building a Docker image. It might take a while.')
+    run_process(f'docker build {resolve(project_dir)} --tag {tag}')
+
+
+def build_dags(
+        root_package,
+        project_dir,
+        docker_repository,
+        start_time,
+        version,
+        specific_workflow=None):
+    for workflow in walk_workflows(root_package):
+        print(f'Generating DAG for {workflow.workflow_id}')
+        if specific_workflow is not None and specific_workflow != workflow.workflow_id:
```

Comment:
if specific_workflow is None and specific_workflow == workflow.workflow_id:
          generate_dag_file(
            resolve(project_dir),
            docker_repository,
            workflow,
            start_time,
            version,
            resolve(root_package).split(os.sep)[-1])

?

Reply:
When you specify a single workflow to generate the DAG, this if statement takes care of skipping the DAGs you didn't choose.